### PR TITLE
Update install manual for cmake flags

### PR DESCRIPTION
**Pull request by sunqm**

### CODE REVIEW

File: source/install.rst [hunk 1]
```diff
@@ -176,18 +176,64 @@ directory can be used to fix this problem::
...
+---------------------------------
+A complete build by default may take long time to finish compilation. In certain
+old operation systems, the build may fail to build `XCFun` if the system does
+not have a proper c++ compiler. To speed up compilation or bypass failed
+compilation, CMake options listed below can be used.
+
+================= ==============================================================
+Flags             Comments
+================= ==============================================================
+`ENABLE_LIBXC`    Whehter to use `LibXC` library in PySCF. If `-DENABLE_XCFUN-OFF`
```
Comment (verena-neufeld) [inline source/install.rst]: Only had a very brief look at this. Noticed "Whehter" -> "Whether" :)

File: source/install.rst [hunk 2]
```diff
@@ -186,17 +186,17 @@ compilation, CMake options listed below can be used.
 ================= ==============================================================
 Flags             Comments
 ================= ==============================================================
-`ENABLE_LIBXC`    Whehter to use `LibXC` library in PySCF. If `-DENABLE_XCFUN-OFF`
+`ENABLE_LIBXC`    Whether to use `LibXC` library in PySCF. If `-DENABLE_XCFUN=OFF`
```
Comment (susilehtola) [inline source/install.rst]: I'm not sure I understand the second sentence: how does XCFun enter into this?

File: source/install.rst [hunk 3]
```diff
@@ -176,18 +176,64 @@ directory can be used to fix this problem::
...
+
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ffast-math -mtune=native -march=native")
+
+Other settings, variables, and flags can also be set in this file::
+
+  set(ENABLE_XCFUN Off)
+  set(WITH_F12 Off)
+
+Some examples of platform-specific configurations can be found in
+directory ``pyscf/lib/cmake_arch_config``.
```
Comment (verena-neufeld) [inline source/install.rst]: I believe this has changed to ``pyscf/lib/cmake_user_inc_examples``

**Comment by verena-neufeld** [comment]
I really like the default column!

File: source/install.rst [hunk 4]
```diff
@@ -176,18 +176,64 @@ directory can be used to fix this problem::
   ``pyscf/lib`` and ``pyscf/lib/deps/lib`` in ``LD_LIBRARY_PATH``.
 
 
+CMake options and compiling flags
+---------------------------------
+A complete build by default may take long time to finish compilation. In certain
+old operation systems, the build may fail to build `XCFun` if the system does
+not have a proper c++ compiler. To speed up compilation or bypass failed
+compilation, CMake options listed below can be used.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
A full build of PySCF may take a long time to finish.
`XCFun` may fail to build a proper C++ compiler is not available, such as on certain old operating systems.
The CMake options listed below can be used to speed up compilation or omit extensions that fail to compile.
```

File: source/install.rst [hunk 5]
```diff
@@ -176,18 +176,64 @@ directory can be used to fix this problem::
...
+
+================= ======= ==============================================================
+Flags             Default Comments
+================= ======= ==============================================================
+`ENABLE_LIBXC`    ON      Whether to use `LibXC` library in PySCF. If `-DENABLE_XCFUN=OFF`
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
`ENABLE_LIBXC`    ON      Whether to use `LibXC` library in PySCF. If `-DENABLE_LIBXC=OFF`
```

File: source/install.rst [hunk 6]
```diff
@@ -176,18 +176,63 @@ directory can be used to fix this problem::
...
+Flags             Default Comments
+================= ======= ==============================================================
+`ENABLE_LIBXC`    ON      Whether to use `LibXC` library in PySCF. If `-DENABLE_LIBXC=OFF`
+                          is appended to cmake command, `LibXC` will not be compiled.
+`ENABLE_XCFUN`    ON      Whether to use `XCFun` library in PySCF. If both
+                          `-DENABLE_LIBXC=OFF` and `-DENABLE_XCFUN=OFF` are set,
+                          importing dft module will lead to `ImportError`.
+`BUILD_LIBXC`     ON      Set it to `OFF` to skip compiling `Libxc`. The dft module
+                          still calls `LibXC` library by default. The dft module will be
+                          linked against the `LibXC` library from early build.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
                          linked against the `LibXC` library from an earlier build.
```

File: source/install.rst [hunk 7]
```diff
@@ -176,18 +176,63 @@ directory can be used to fix this problem::
...
+The CMake options listed below can be used to speed up compilation or omit extensions that fail to compile.
+
+================= ======= ==============================================================
+Flags             Default Comments
+================= ======= ==============================================================
+`ENABLE_LIBXC`    ON      Whether to use `LibXC` library in PySCF. If `-DENABLE_LIBXC=OFF`
+                          is appended to cmake command, `LibXC` will not be compiled.
+`ENABLE_XCFUN`    ON      Whether to use `XCFun` library in PySCF. If both
+                          `-DENABLE_LIBXC=OFF` and `-DENABLE_XCFUN=OFF` are set,
+                          importing dft module will lead to `ImportError`.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
`ENABLE_XCFUN`    ON      Whether to use `XCFun` library in PySCF. If `-DENABLE_XCFUN=OFF`
                          is appended to cmake command, `XCFun` will not be compiled.
```

File: source/install.rst [hunk 8]
```diff
@@ -176,18 +176,63 @@ directory can be used to fix this problem::
...
+`BUILD_LIBXC`     ON      Set it to `OFF` to skip compiling `Libxc`. The dft module
+                          still calls `LibXC` library by default. The dft module will be
+                          linked against the `LibXC` library from an earlier build.
+`BUILD_XCFUN`     ON      Set it to `OFF` to skip compiling `XCFun`. The dft module
+                          will be linked against the `XCFun` library from early build.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
                          will be linked against the `XCFun` library from an earlier build.
```

File: source/install.rst [hunk 9]
```diff
@@ -176,18 +176,62 @@ directory can be used to fix this problem::
...
+                          is appended to cmake command, `LibXC` will not be compiled.
+`ENABLE_XCFUN`    ON      Whether to use `XCFun` library in PySCF. If `-DENABLE_XCFUN=OFF`
+                          is appended to cmake command, `XCFun` will not be compiled.
+`BUILD_LIBXC`     ON      Set it to `OFF` to skip compiling `Libxc`. The dft module
+                          still calls `LibXC` library by default. The dft module will be
+                          linked against the `LibXC` library from an earlier build.
+`BUILD_XCFUN`     ON      Set it to `OFF` to skip compiling `XCFun`. The dft module
+                          will be linked against the `XCFun` library from an earlier build.
+`BUILD_LIBCINT`   ON      Set it to `OFF` to skip compiling `libcint`. The integral
+                          library from early build will be used.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
                          library from an earlier build will be used.
```

File: source/install.rst [hunk 10]
```diff
@@ -176,18 +176,62 @@ directory can be used to fix this problem::
...
+`ENABLE_XCFUN`    ON      Whether to use `XCFun` library in PySCF. If `-DENABLE_XCFUN=OFF`
+                          is appended to cmake command, `XCFun` will not be compiled.
+`BUILD_LIBXC`     ON      Set it to `OFF` to skip compiling `Libxc`. The dft module
+                          still calls `LibXC` library by default. The dft module will be
+                          linked against the `LibXC` library from an earlier build.
+`BUILD_XCFUN`     ON      Set it to `OFF` to skip compiling `XCFun`. The dft module
+                          will be linked against the `XCFun` library from an earlier build.
+`BUILD_LIBCINT`   ON      Set it to `OFF` to skip compiling `libcint`. The integral
+                          library from an earlier build will be used.
+`WITH_F12`        ON      Whether or not to compile the F12 relevant integrals.
```
Comment (susilehtola) [inline source/install.rst]:
```suggestion
`WITH_F12`        ON      Whether to compile the F12 integrals.
```

File: source/install.rst [hunk 11]
```diff
@@ -176,18 +176,62 @@ directory can be used to fix this problem::
   ``pyscf/lib`` and ``pyscf/lib/deps/lib`` in ``LD_LIBRARY_PATH``.
 
 
+CMake options and compiling flags
+---------------------------------
+A full build of PySCF may take a long time to finish.
+`XCFun` may fail to build a proper C++ compiler is not available, such as on certain old operating systems.
+The CMake options listed below can be used to speed up compilation or omit extensions that fail to compile.
+
```
Comment (susilehtola) [inline source/install.rst:185]:
```suggestion
Note:  If both `-DENABLE_LIBXC=OFF` and `-DENABLE_XCFUN=OFF` are set, importing the dft module will lead to an `ImportError`.

```

**Review by susilehtola — APPROVED** [review]

**Pull request merged**